### ISSUE TITLE
Subprocess commands missing from history

### ISSUE THREAD

**Issue by tjarvstrand** [atuinsh/atuin#1847]
Hi,

First, thanks for a great tool! It's been a joy to use so far.

I routinely do eg. `(cd x; mycommand)` when I only want to run one or two commands inside a directory and I've noticed that these don't end up in the history. Is there any way I can get them to do that?

**Comment by arcuru** [comment]
Interesting, they show up for me using either bash or zsh. What's your environment like?

Note that the whole command is what will be in history, `(cd x; mycommand)`, and the logged directory will be where you ran _that_ command. I'm not sure if you're asking about `mycommand` itself not showing up.

**Comment by tjarvstrand** [comment]
Yeah, `(cd x; mycommand)` is what I would like to be in the history :)

I'm on bash `5.2.15` on MacOS Ventura.

I do get the commands in the history, i.e. I can use the up arrow to run the command, but when I hit ctrl-r, they don't appear in that list.

**Comment by kenjo** [comment]
don't show up for me
GNU bash, version 5.2.15(1)-release (x86_64-pc-linux-gnu)

there also is a problem if I use c-x c-e. I do have 
shopt -s lithist
but removing that has the same result

using c-p I get the correct history in the built in list.

**Comment by frasty** [comment]
> Same here, e.g., `(true)` does not end up in `atuin`'s history
> 
> ```
> atuin:
>   version: 18.1.0
>   sync: null
> shell:
>   name: bash
>   plugins:
>   - atuin
>   - bash-preexec
> system:
>   os: Fedora Linux
>   arch: x86_64
>   version: '38'
> ```
> 
> To be honest, I haven't seen the `atuin` source code, but I'm very curious why there are all sort of such weird "missing in history" issues. Do you do some "parsing" or whatever on history commands? If so, could you explain why? And if so, is there a chance we can have our history intact (optionally, of course, some may not like it), matching exactly what's in our shell history? Thanks.

I definitely second this. Sometimes I am unwilling to search history through atuin because I know that some commands could be missed from history. I'd rather prefer to have my history intact even if that could lead to some privacy issues.

**Comment by akinomyoga** [comment]
> why there are all sort of such weird "missing in history" issues.

@mjf @frasty Those are all the limitations of `bash-preexec`, which Atuin uses to hook into the command execution of Bash.

> Do you do some "parsing" or whatever on history commands?

No. Atuin maintains a separate command history, which is independent of the command history of Bash. When Atuin shows the search menu, it doesn't access the command history of Bash.

> is there a chance we can have our history intact (optionally, of course, some may not like it), matching exactly what's in our shell history?

Have you noticed [the Bash section](https://github.com/atuinsh/atuin/#bash) in README? Currently, ble.sh is the only option if you want to record all the commands by Atuin. I think those descriptions are easy to miss, so I submitted a PR #1937.

**Comment by ellie** [comment]
In addition, there are no such errors with zsh/fish/xonsh/nushell. This is purely because Bash provides absolutely no way for us to hook into the shell lifecycle.

Bash preexec mostly just hacks the prompt. It works well enough, but there are some issues - as you've noted. 

@akinomyoga's ble.sh is a reimplementation of the line editor in Bash, so has far more control and works better. We cannot install this by default however, as it does much more than just add some shell hooks

You'll probably find that a bunch of other shell plugins don't function quite so well with Bash

**Comment by ellie** [comment]
@mjf you can call `atuin history start` and  `atuin history end` manually yourself, if you wish. Check out the bash integration to see how it's used.

**Comment by ellie** [comment]
Closing as documented expected behaviour. This is a bash-preexec limitation, see the Shell Integration docs. Use ble.sh as the bash preexec backend if you need complete capture.

> The simple atuin history put I suggest above, would likely solve it for my needs. Or, do I miss something here?

Running `history start/history end` will do the same as a single `put` would, so I'm not sure this is required
